website/notifications/utils.py

```diff
@@ -460,6 +460,9 @@ def subscribe_user_to_notifications(node, user):
     if node.is_deleted:
         raise InvalidSubscriptionError('Deleted Nodes are invalid targets for subscriptions')
 
+    if getattr(node, 'is_registration', False):
+        raise InvalidSubscriptionError('Registrations are invalid targets for subscriptions')
```

Comment:
Throwing an error if a registration is passed into `subscribe_user_to_notifications`.  This is called as part of a project being created or a contributor being added.  As far as project creation, the signal is only being sent if it's a node, and you can't add contributors to a registration.  So I'm not sure how this would be hit normally, but this check should help.